### ISSUE TITLE
🐛 `useImportExtensions` no longer highlights imports without extension

### ISSUE THREAD

**Issue by mintydev789** [biomejs/biome#8694]
### Environment information

<details>

```bash
CLI:
  Version:                      2.3.11
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         alacritty
  JS_RUNTIME_VERSION:           v24.11.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.6.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false
  HTML full support enabled:    unset

Workspace:
  Open Documents:               0
```

</details>

### What happened?

1. Used the following config file:

```json
{
  "$schema": "https://biomejs.dev/schemas/2.3.11/schema.json",
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true,
      "correctness": {
        "useImportExtensions": "warn"
      }
    }
  }
}
```

2. Opened one of the TypeScript files where Biome is acting as linter (I checked this is the case) and edited one of the imports to have no `.ts` extension at the end.
3. Linter does not highlight the line as warning. I believe this worked in a previous version of Biome.

### Expected result

The linter correctly highlights the line without an extension AND applies the fix on save (assuming automatic fixes are enabled).

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by mintydev789** [comment]
> Hello @mintydev789, please provide a minimal reproduction. You can use one of the following options:
> 
> - Provide a link to [our playground](https://biomejs.dev/playground), if it's applicable.
> - Provide a link to GitHub repository. To easily create a reproduction, you can use our interactive CLI via `npm create @biomejs/biome-reproduction`
> 
> Issues marked with `S-Needs repro` will be **closed** if they have **no activity within 3 days**.

I already provided a reproduction in the issue.

**Comment by siketyan** [comment]
@mintydev789 Please create a reproduction in either way the bot suggests so we can investigate it easily.

**Comment by mintydev789** [comment]
> @mintydev789 Please create a reproduction in either way the bot suggests so we can investigate it easily.

Sorry, but I'm deep in a big code refactor update right now and don't have the time to do this. The issue should have everything needed to reproduce this.

**Comment by ematipico** [comment]
> The issue should have everything needed to reproduce this.

I just tried using your instructions and the rule is working as expected (using 2.4.10). Provide a valid reproduction.
